

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Align Reads (YJF153) to the Reference (S288c) &mdash; Project: Calling SNPs in Yeast strain YJF153 1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Index Genome using Samtools" href="32_genome_sort_index.html" />
    <link rel="prev" title="Index Genome using Burrows-Wheeler Aligner (BWA)" href="22_genome_indexing.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> Project: Calling SNPs in Yeast strain YJF153
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Section 01: Data Download</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="00_check_environment.html">Check if the environment is correctly setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="01_fastq_download.html">Download SRA Data: SRR4841864</a></li>
<li class="toctree-l1"><a class="reference internal" href="02_fastq_exploration.html">Explore the data</a></li>
<li class="toctree-l1"><a class="reference internal" href="03_fastq_generator.html">Generator/Stream of FASTQ reads</a></li>
</ul>
<p class="caption"><span class="caption-text">Section 02: Data Preprocessing</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="10_intro2quality_score.html">Quality Score in High-throughput Sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="11_quality_score_raw.html">Quality score of raw reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="12_fastqc_raw.html">FASTQC Report of Raw Reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="13_adapter.html">Find the adapter</a></li>
<li class="toctree-l1"><a class="reference internal" href="14_trim_adapters.html">Trimming Adapters of the reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="15_filter_reads.html">Filter Reads: Select reads with high quality</a></li>
<li class="toctree-l1"><a class="reference internal" href="16_fastqc_prep.html">FASTQC Report of Trimmed/Filtered (Preprocessed) Reads</a></li>
</ul>
<p class="caption"><span class="caption-text">Section 03: Download and Indexing Genome</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="21_genome_download.html">Download Yeast Genome (Strain S288c)</a></li>
<li class="toctree-l1"><a class="reference internal" href="22_genome_indexing.html">Index Genome using Burrows-Wheeler Aligner (BWA)</a></li>
</ul>
<p class="caption"><span class="caption-text">Section 04: Alignment and calls of SNPs</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Align Reads (YJF153) to the Reference (S288c)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#Alignment-using-BWA-mem">Alignment using <code class="docutils literal notranslate"><span class="pre">BWA</span> <span class="pre">mem</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="32_genome_sort_index.html">Index Genome using Samtools</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Project: Calling SNPs in Yeast strain YJF153</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Align Reads (YJF153) to the Reference (S288c)</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/31_alignment.ipynb.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 5ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="Align-Reads-(YJF153)-to-the-Reference-(S288c)">
<h1>Align Reads (YJF153) to the Reference (S288c)<a class="headerlink" href="#Align-Reads-(YJF153)-to-the-Reference-(S288c)" title="Permalink to this headline">¶</a></h1>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">source</span> config.sh
</pre></div>
</div>
</div>
<p>check the files</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls <span class="nv">$RAW_FASTQS</span>
ls <span class="nv">$TRIMMED</span>
ls <span class="nv">$FILTER</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
SRR4841864_1.fastq  SRR4841864_2.fastq  SRR4841864.sra
SRR4841864_1.trim.fastq  SRR4841864_2.trim.fastq
SRR4841864_1.filter.fastq  SRR4841864_2.filter.fastq
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>ls <span class="nv">$GENOME</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa
Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.amb
Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.ann
Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.bwt
<span class="ansi-red-intense-fg ansi-bold">Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.gz</span>
Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.pac
Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.sa
</pre></div></div>
</div>
<div class="section" id="Alignment-using-BWA-mem">
<h2>Alignment using <code class="docutils literal notranslate"><span class="pre">BWA</span> <span class="pre">mem</span></code><a class="headerlink" href="#Alignment-using-BWA-mem" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://github.com/lh3/bwa">https://github.com/lh3/bwa</a></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">lh3</span><span class="o">/</span><span class="n">bwa</span><span class="o">.</span><span class="n">git</span>
<span class="n">cd</span> <span class="n">bwa</span><span class="p">;</span> <span class="n">make</span>
<span class="o">./</span><span class="n">bwa</span> <span class="n">index</span> <span class="n">ref</span><span class="o">.</span><span class="n">fa</span>
<span class="o">./</span><span class="n">bwa</span> <span class="n">mem</span> <span class="n">ref</span><span class="o">.</span><span class="n">fa</span> <span class="n">read</span><span class="o">-</span><span class="n">se</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> <span class="o">|</span> <span class="n">gzip</span> <span class="o">-</span><span class="mi">3</span> <span class="o">&gt;</span> <span class="n">aln</span><span class="o">-</span><span class="n">se</span><span class="o">.</span><span class="n">sam</span><span class="o">.</span><span class="n">gz</span>
<span class="o">./</span><span class="n">bwa</span> <span class="n">mem</span> <span class="n">ref</span><span class="o">.</span><span class="n">fa</span> <span class="n">read1</span><span class="o">.</span><span class="n">fq</span> <span class="n">read2</span><span class="o">.</span><span class="n">fq</span> <span class="o">|</span> <span class="n">gzip</span> <span class="o">-</span><span class="mi">3</span> <span class="o">&gt;</span> <span class="n">aln</span><span class="o">-</span><span class="n">pe</span><span class="o">.</span><span class="n">sam</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p>Parameters</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">-</span><span class="n">t</span> <span class="n">INT</span> <span class="o">---&gt;</span> <span class="n">Number</span> <span class="n">of</span> <span class="n">threads</span> <span class="p">[</span><span class="mi">1</span><span class="p">]</span>
<span class="o">-</span><span class="n">M</span>     <span class="o">---&gt;</span> <span class="n">Mark</span> <span class="n">shorter</span> <span class="n">split</span> <span class="n">hits</span> <span class="k">as</span> <span class="n">secondary</span> <span class="p">(</span><span class="k">for</span> <span class="n">Picard</span> <span class="n">compatibility</span><span class="p">)</span><span class="o">.</span>
</pre></div>
</div>
<p><strong>Test looping in/out directories and fastq files</strong></p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nv">DIR</span><span class="o">=(</span> <span class="s2">&quot;</span><span class="nv">$RAW_FASTQS</span><span class="s2">&quot;</span> <span class="s2">&quot;</span><span class="nv">$TRIMMED</span><span class="s2">&quot;</span>    <span class="s2">&quot;</span><span class="nv">$FILTER</span><span class="s2">&quot;</span> <span class="o">)</span>
<span class="nv">OUT</span><span class="o">=(</span> <span class="s2">&quot;</span><span class="nv">$ALIGN_RAW</span><span class="s2">&quot;</span>  <span class="s2">&quot;</span><span class="nv">$ALIGN_TRIM</span><span class="s2">&quot;</span> <span class="s2">&quot;</span><span class="nv">$ALIGN_FILTER</span><span class="s2">&quot;</span> <span class="o">)</span>
<span class="nv">REFFA</span><span class="o">=</span><span class="nv">$GENOME</span>/<span class="nv">$FA</span>

<span class="k">for</span> <span class="o">((</span><span class="nv">i</span><span class="o">=</span><span class="m">0</span><span class="p">;</span> i&lt;<span class="si">${#</span><span class="nv">OUT</span><span class="p">[@]</span><span class="si">}</span><span class="p">;</span> ++i<span class="o">))</span><span class="p">;</span> <span class="k">do</span>
    <span class="nb">echo</span> <span class="s2">&quot;================================================&quot;</span>
    <span class="nv">READ1</span><span class="o">=</span><span class="si">${</span><span class="nv">DIR</span><span class="p">[i]</span><span class="si">}</span>/SRR4841864_1*.fastq
    <span class="nv">READ2</span><span class="o">=</span><span class="si">${</span><span class="nv">DIR</span><span class="p">[i]</span><span class="si">}</span>/SRR4841864_2*.fastq

    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">DIR</span><span class="p">[i]</span><span class="si">}</span><span class="s2">&quot;</span>
    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">OUT</span><span class="p">[i]</span><span class="si">}</span><span class="s2">&quot;</span>
    ls <span class="nv">$READ1</span>
    ls <span class="nv">$READ2</span>
    <span class="nb">echo</span> <span class="s2">&quot;++++++++++++++++++++++++++++++++++++++++++++++++&quot;</span>
<span class="k">done</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
================================================
/home/jovyan/work/Data/SRR4841864/rawdata
/home/jovyan/work/Data/SRR4841864/align_raw
/home/jovyan/work/Data/SRR4841864/rawdata/SRR4841864_1.fastq
/home/jovyan/work/Data/SRR4841864/rawdata/SRR4841864_2.fastq
++++++++++++++++++++++++++++++++++++++++++++++++
================================================
/home/jovyan/work/Data/SRR4841864/fastqs_trimmed
/home/jovyan/work/Data/SRR4841864/align_trim
/home/jovyan/work/Data/SRR4841864/fastqs_trimmed/SRR4841864_1.trim.fastq
/home/jovyan/work/Data/SRR4841864/fastqs_trimmed/SRR4841864_2.trim.fastq
++++++++++++++++++++++++++++++++++++++++++++++++
================================================
/home/jovyan/work/Data/SRR4841864/fastqs_filter
/home/jovyan/work/Data/SRR4841864/align_filter
/home/jovyan/work/Data/SRR4841864/fastqs_filter/SRR4841864_1.filter.fastq
/home/jovyan/work/Data/SRR4841864/fastqs_filter/SRR4841864_2.filter.fastq
++++++++++++++++++++++++++++++++++++++++++++++++
</pre></div></div>
</div>
<p><strong>Alignment raw, trimmed, and filtered reads to the reference</strong></p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nv">DIR</span><span class="o">=(</span> <span class="s2">&quot;</span><span class="nv">$RAW_FASTQS</span><span class="s2">&quot;</span> <span class="s2">&quot;</span><span class="nv">$TRIMMED</span><span class="s2">&quot;</span>    <span class="s2">&quot;</span><span class="nv">$FILTER</span><span class="s2">&quot;</span> <span class="o">)</span>
<span class="nv">OUT</span><span class="o">=(</span> <span class="s2">&quot;</span><span class="nv">$ALIGN_RAW</span><span class="s2">&quot;</span>  <span class="s2">&quot;</span><span class="nv">$ALIGN_TRIM</span><span class="s2">&quot;</span> <span class="s2">&quot;</span><span class="nv">$ALIGN_FILTER</span><span class="s2">&quot;</span> <span class="o">)</span>
<span class="nv">REFFA</span><span class="o">=</span><span class="nv">$GENOME</span>/<span class="nv">$FA</span>

<span class="k">for</span> <span class="o">((</span><span class="nv">i</span><span class="o">=</span><span class="m">0</span><span class="p">;</span> i&lt;<span class="si">${#</span><span class="nv">OUT</span><span class="p">[@]</span><span class="si">}</span><span class="p">;</span> ++i<span class="o">))</span><span class="p">;</span> <span class="k">do</span>
    <span class="nb">echo</span> <span class="s2">&quot;================================================&quot;</span>
    <span class="nv">READ1</span><span class="o">=</span><span class="si">${</span><span class="nv">DIR</span><span class="p">[i]</span><span class="si">}</span>/SRR4841864_1*.fastq
    <span class="nv">READ2</span><span class="o">=</span><span class="si">${</span><span class="nv">DIR</span><span class="p">[i]</span><span class="si">}</span>/SRR4841864_2*.fastq

    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">DIR</span><span class="p">[i]</span><span class="si">}</span><span class="s2">&quot;</span>
    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">OUT</span><span class="p">[i]</span><span class="si">}</span><span class="s2">&quot;</span>
    ls <span class="nv">$READ1</span>
    ls <span class="nv">$READ2</span>
    <span class="nb">echo</span> <span class="s2">&quot;++++++++++++++++++++++++++++++++++++++++++++++++&quot;</span>
    bwa mem -t <span class="m">10</span> <span class="nv">$REFFA</span> <span class="nv">$READ1</span> <span class="nv">$READ2</span> &gt; <span class="si">${</span><span class="nv">OUT</span><span class="p">[i]</span><span class="si">}</span>/aln_pe.sam
<span class="k">done</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
================================================
/home/jovyan/work/Data/SRR4841864/rawdata
/home/jovyan/work/Data/SRR4841864/align_raw
/home/jovyan/work/Data/SRR4841864/rawdata/SRR4841864_1.fastq
/home/jovyan/work/Data/SRR4841864/rawdata/SRR4841864_2.fastq
++++++++++++++++++++++++++++++++++++++++++++++++
[M::bwa_idx_load_from_disk] read 0 ALT contigs
[M::process] read 990100 sequences (100000100 bp)...
[M::process] read 990100 sequences (100000100 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 426894, 34, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (222, 290, 352)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 612)
[M::mem_pestat] mean and std.dev: (289.16, 95.30)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 742)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (131, 302, 1100)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3038)
[M::mem_pestat] mean and std.dev: (607.18, 611.24)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4007)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 990100 reads in 58.622 CPU sec, 5.931 real sec
[M::process] read 990100 sequences (100000100 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 426838, 28, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (220, 288, 350)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 610)
[M::mem_pestat] mean and std.dev: (286.60, 94.98)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 740)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (134, 348, 1081)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2975)
[M::mem_pestat] mean and std.dev: (551.08, 485.27)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 3922)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 990100 reads in 62.321 CPU sec, 6.241 real sec
[M::process] read 990100 sequences (100000100 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 426800, 28, 1)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (218, 286, 349)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 611)
[M::mem_pestat] mean and std.dev: (285.36, 94.91)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 742)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (171, 302, 1144)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3090)
[M::mem_pestat] mean and std.dev: (645.75, 646.09)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4063)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 990100 reads in 61.640 CPU sec, 6.158 real sec
[M::process] read 990100 sequences (100000100 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 426486, 34, 1)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (221, 289, 352)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 614)
[M::mem_pestat] mean and std.dev: (288.33, 95.66)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 745)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (166, 345, 997)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2659)
[M::mem_pestat] mean and std.dev: (565.09, 556.41)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 3490)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 990100 reads in 61.377 CPU sec, 6.138 real sec
[M::process] read 990100 sequences (100000100 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 426995, 24, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (219, 287, 349)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 609)
[M::mem_pestat] mean and std.dev: (285.94, 95.00)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 739)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (150, 202, 1041)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2823)
[M::mem_pestat] mean and std.dev: (381.52, 400.68)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 3714)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 990100 reads in 61.706 CPU sec, 6.204 real sec
[M::process] read 990100 sequences (100000100 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 427885, 25, 1)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (223, 291, 353)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 613)
[M::mem_pestat] mean and std.dev: (290.06, 95.92)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 743)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (103, 189, 606)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1612)
[M::mem_pestat] mean and std.dev: (324.58, 359.49)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 2115)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 990100 reads in 60.751 CPU sec, 6.071 real sec
[M::process] read 990100 sequences (100000100 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 428920, 41, 2)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (227, 295, 357)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 617)
[M::mem_pestat] mean and std.dev: (293.85, 96.36)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 747)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (139, 393, 1155)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3187)
[M::mem_pestat] mean and std.dev: (669.69, 626.07)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4203)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 990100 reads in 59.787 CPU sec, 5.980 real sec
[M::process] read 990100 sequences (100000100 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 429132, 34, 2)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (226, 293, 356)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 616)
[M::mem_pestat] mean and std.dev: (292.35, 96.44)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 746)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (83, 162, 1141)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3257)
[M::mem_pestat] mean and std.dev: (498.84, 539.72)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4315)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 990100 reads in 59.027 CPU sec, 5.885 real sec
[M::process] read 990100 sequences (100000100 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 427903, 31, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (226, 294, 357)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 619)
[M::mem_pestat] mean and std.dev: (293.39, 96.87)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 750)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (194, 307, 1104)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2924)
[M::mem_pestat] mean and std.dev: (601.10, 514.33)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 3834)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 990100 reads in 59.399 CPU sec, 5.919 real sec
[M::process] read 882444 sequences (89126844 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 428053, 31, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (226, 294, 357)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 619)
[M::mem_pestat] mean and std.dev: (293.24, 96.84)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 750)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (132, 217, 1094)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3018)
[M::mem_pestat] mean and std.dev: (534.48, 625.03)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 3980)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 990100 reads in 58.646 CPU sec, 5.864 real sec
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 381695, 28, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (224, 292, 355)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 617)
[M::mem_pestat] mean and std.dev: (291.09, 96.65)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 748)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (108, 249, 1085)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3039)
[M::mem_pestat] mean and std.dev: (500.00, 511.66)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4016)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 882444 reads in 52.038 CPU sec, 5.247 real sec
[main] Version: 0.7.17-r1198-dirty
[main] CMD: bwa mem -t 10 /home/jovyan/work/Data/SRR4841864/genome/Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa /home/jovyan/work/Data/SRR4841864/rawdata/SRR4841864_1.fastq /home/jovyan/work/Data/SRR4841864/rawdata/SRR4841864_2.fastq
[main] Real time: 69.654 sec; CPU: 656.981 sec
================================================
/home/jovyan/work/Data/SRR4841864/fastqs_trimmed
/home/jovyan/work/Data/SRR4841864/align_trim
/home/jovyan/work/Data/SRR4841864/fastqs_trimmed/SRR4841864_1.trim.fastq
/home/jovyan/work/Data/SRR4841864/fastqs_trimmed/SRR4841864_2.trim.fastq
++++++++++++++++++++++++++++++++++++++++++++++++
[M::bwa_idx_load_from_disk] read 0 ALT contigs
[M::process] read 1009544 sequences (100000165 bp)...
[M::process] read 1012214 sequences (100000075 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 439051, 32, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (221, 288, 350)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 608)
[M::mem_pestat] mean and std.dev: (286.96, 94.30)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 737)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (131, 263, 1145)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3173)
[M::mem_pestat] mean and std.dev: (622.09, 628.84)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4187)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 1009544 reads in 56.704 CPU sec, 5.718 real sec
[M::process] read 1010692 sequences (100000142 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 439537, 30, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (217, 285, 346)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 604)
[M::mem_pestat] mean and std.dev: (283.64, 93.83)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 733)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (119, 316, 999)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2759)
[M::mem_pestat] mean and std.dev: (482.55, 465.24)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 3639)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 1012214 reads in 60.097 CPU sec, 6.015 real sec
[M::process] read 1008818 sequences (100000146 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 438834, 29, 2)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (217, 285, 347)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 607)
[M::mem_pestat] mean and std.dev: (283.84, 93.98)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 737)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (213, 619, 1219)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3231)
[M::mem_pestat] mean and std.dev: (828.41, 710.84)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4237)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 1010692 reads in 59.354 CPU sec, 5.937 real sec
[M::process] read 1010238 sequences (100000139 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 437779, 34, 1)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (219, 287, 348)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 606)
[M::mem_pestat] mean and std.dev: (285.51, 94.59)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 735)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (124, 253, 617)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1603)
[M::mem_pestat] mean and std.dev: (404.55, 381.01)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 2096)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 1008818 reads in 58.838 CPU sec, 5.896 real sec
[M::process] read 1006472 sequences (100000008 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 438543, 25, 1)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (216, 284, 345)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 603)
[M::mem_pestat] mean and std.dev: (282.58, 93.67)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 732)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (87, 202, 1111)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3159)
[M::mem_pestat] mean and std.dev: (482.20, 496.93)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4183)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 1010238 reads in 60.009 CPU sec, 5.982 real sec
[M::process] read 1006094 sequences (100000029 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 437928, 28, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (226, 293, 355)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 613)
[M::mem_pestat] mean and std.dev: (291.80, 95.69)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 742)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (118, 210, 1066)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2962)
[M::mem_pestat] mean and std.dev: (443.04, 441.59)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 3910)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 1006472 reads in 57.350 CPU sec, 5.714 real sec
[M::process] read 1005114 sequences (100000039 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 438289, 37, 3)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (226, 293, 355)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 613)
[M::mem_pestat] mean and std.dev: (292.12, 95.70)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 742)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (140, 319, 1239)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3437)
[M::mem_pestat] mean and std.dev: (715.77, 710.13)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4536)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 1006094 reads in 58.409 CPU sec, 5.836 real sec
[M::process] read 1005350 sequences (100000056 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 437521, 37, 1)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (224, 292, 354)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 614)
[M::mem_pestat] mean and std.dev: (290.97, 96.03)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 744)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (83, 252, 1183)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3383)
[M::mem_pestat] mean and std.dev: (513.23, 554.35)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4483)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 1005114 reads in 57.075 CPU sec, 5.651 real sec
[M::process] read 1005670 sequences (100000177 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 437162, 29, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (226, 294, 356)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 616)
[M::mem_pestat] mean and std.dev: (292.65, 96.25)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 746)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (102, 275, 1121)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3159)
[M::mem_pestat] mean and std.dev: (548.10, 509.54)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4178)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 1005350 reads in 58.593 CPU sec, 5.873 real sec
[M::process] read 383322 sequences (38165889 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 437115, 32, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (224, 292, 354)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 614)
[M::mem_pestat] mean and std.dev: (291.02, 96.10)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 744)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (120, 260, 1146)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3198)
[M::mem_pestat] mean and std.dev: (590.16, 649.69)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4224)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 1005670 reads in 59.286 CPU sec, 5.915 real sec
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 166715, 18, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (222, 290, 353)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 615)
[M::mem_pestat] mean and std.dev: (289.15, 96.12)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 746)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (108, 170, 997)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2775)
[M::mem_pestat] mean and std.dev: (408.33, 428.89)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 3664)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 383322 reads in 22.809 CPU sec, 2.294 real sec
[main] Version: 0.7.17-r1198-dirty
[main] CMD: bwa mem -t 10 /home/jovyan/work/Data/SRR4841864/genome/Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa /home/jovyan/work/Data/SRR4841864/fastqs_trimmed/SRR4841864_1.trim.fastq /home/jovyan/work/Data/SRR4841864/fastqs_trimmed/SRR4841864_2.trim.fastq
[main] Real time: 65.708 sec; CPU: 609.903 sec
================================================
/home/jovyan/work/Data/SRR4841864/fastqs_filter
/home/jovyan/work/Data/SRR4841864/align_filter
/home/jovyan/work/Data/SRR4841864/fastqs_filter/SRR4841864_1.filter.fastq
/home/jovyan/work/Data/SRR4841864/fastqs_filter/SRR4841864_2.filter.fastq
++++++++++++++++++++++++++++++++++++++++++++++++
[M::bwa_idx_load_from_disk] read 0 ALT contigs
[M::process] read 991408 sequences (100000031 bp)...
[M::process] read 991376 sequences (100000117 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 439427, 25, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (215, 281, 341)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 593)
[M::mem_pestat] mean and std.dev: (279.74, 91.20)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 719)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (128, 210, 1070)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2954)
[M::mem_pestat] mean and std.dev: (577.08, 653.54)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 3896)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 991408 reads in 55.528 CPU sec, 5.600 real sec
[M::process] read 991374 sequences (100000072 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 439712, 31, 1)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (212, 278, 338)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 590)
[M::mem_pestat] mean and std.dev: (276.97, 90.74)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 716)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (171, 302, 1189)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3225)
[M::mem_pestat] mean and std.dev: (694.77, 730.74)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4243)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 991376 reads in 56.316 CPU sec, 5.636 real sec
[M::process] read 991452 sequences (100000195 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 439237, 25, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (211, 278, 338)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 592)
[M::mem_pestat] mean and std.dev: (276.97, 91.12)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 719)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (150, 225, 1113)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3039)
[M::mem_pestat] mean and std.dev: (476.92, 443.11)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4002)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 991374 reads in 57.122 CPU sec, 5.713 real sec
[M::process] read 991454 sequences (100000098 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 438170, 23, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (221, 287, 348)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 602)
[M::mem_pestat] mean and std.dev: (285.82, 93.02)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 729)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (124, 256, 1095)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3037)
[M::mem_pestat] mean and std.dev: (561.78, 614.19)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4008)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 991452 reads in 56.268 CPU sec, 5.662 real sec
[M::process] read 991480 sequences (100000012 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 438374, 37, 1)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (223, 290, 350)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 604)
[M::mem_pestat] mean and std.dev: (288.19, 93.41)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 731)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (104, 162, 1101)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3095)
[M::mem_pestat] mean and std.dev: (457.18, 522.24)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4092)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 991454 reads in 55.798 CPU sec, 5.573 real sec
[M::process] read 991392 sequences (100000052 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 437364, 30, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (223, 290, 351)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 607)
[M::mem_pestat] mean and std.dev: (288.44, 94.05)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 735)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (239, 363, 1135)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2927)
[M::mem_pestat] mean and std.dev: (638.00, 522.89)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 3823)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 991480 reads in 55.692 CPU sec, 5.584 real sec
[M::process] read 128318 sequences (12942850 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 437383, 41, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (222, 288, 349)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 603)
[M::mem_pestat] mean and std.dev: (287.29, 93.75)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 730)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (100, 196, 1085)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3055)
[M::mem_pestat] mean and std.dev: (462.95, 552.90)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 4040)
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation RF
[M::mem_process_seqs] Processed 991392 reads in 56.488 CPU sec, 5.684 real sec
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 56311, 5, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (219, 287, 348)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 606)
[M::mem_pestat] mean and std.dev: (285.13, 93.97)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 735)
[M::mem_pestat] skip orientation RF as there are not enough pairs
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_process_seqs] Processed 128318 reads in 8.623 CPU sec, 0.897 real sec
[main] Version: 0.7.17-r1198-dirty
[main] CMD: bwa mem -t 10 /home/jovyan/work/Data/SRR4841864/genome/Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa /home/jovyan/work/Data/SRR4841864/fastqs_filter/SRR4841864_1.filter.fastq /home/jovyan/work/Data/SRR4841864/fastqs_filter/SRR4841864_2.filter.fastq
[main] Real time: 44.834 sec; CPU: 403.063 sec
</pre></div></div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="32_genome_sort_index.html" class="btn btn-neutral float-right" title="Index Genome using Samtools" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="22_genome_indexing.html" class="btn btn-neutral float-left" title="Index Genome using Burrows-Wheeler Aligner (BWA)" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>