

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Call SNP using bcftools &mdash; Project: Calling SNPs in Yeast strain YJF153 1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Sort and Index the alignment results" href="33_sort_index.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> Project: Calling SNPs in Yeast strain YJF153
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Section 01: Data Download</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="00_check_environment.html">Check if the environment is correctly setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="01_fastq_download.html">Download SRA Data: SRR4841864</a></li>
<li class="toctree-l1"><a class="reference internal" href="02_fastq_exploration.html">Explore the data</a></li>
<li class="toctree-l1"><a class="reference internal" href="03_fastq_generator.html">Generator/Stream of FASTQ reads</a></li>
</ul>
<p class="caption"><span class="caption-text">Section 02: Data Preprocessing</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="10_intro2quality_score.html">Quality Score in High-throughput Sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="11_quality_score_raw.html">Quality score of raw reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="12_fastqc_raw.html">FASTQC Report of Raw Reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="13_adapter.html">Find the adapter</a></li>
<li class="toctree-l1"><a class="reference internal" href="14_trim_adapters.html">Trimming Adapters of the reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="15_filter_reads.html">Filter Reads: Select reads with high quality</a></li>
<li class="toctree-l1"><a class="reference internal" href="16_fastqc_prep.html">FASTQC Report of Trimmed/Filtered (Preprocessed) Reads</a></li>
</ul>
<p class="caption"><span class="caption-text">Section 03: Download and Indexing Genome</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="21_genome_download.html">Download Yeast Genome (Strain S288c)</a></li>
<li class="toctree-l1"><a class="reference internal" href="22_genome_indexing.html">Index Genome using Burrows-Wheeler Aligner (BWA)</a></li>
</ul>
<p class="caption"><span class="caption-text">Section 04: Alignment and Calls of SNPs</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="31_alignment.html">Align Reads (YJF153) to the Reference (S288c)</a></li>
<li class="toctree-l1"><a class="reference internal" href="32_genome_sort_index.html">Index Genome using Samtools</a></li>
<li class="toctree-l1"><a class="reference internal" href="33_sort_index.html">Sort and Index the alignment results</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Call SNP using bcftools</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#bcftools-—-utilities-for-variant-calling-and-manipulating-VCFs-and-BCFs.">bcftools — utilities for variant calling and manipulating VCFs and BCFs.</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#bcftools-mpileup-is-used-when-calling-SNP"><strong>bcftools mpileup is used when calling SNP</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#Look-closer-to-mpileup-—-multi-way-pileup-producing-genotype-likelihoods"><strong>Look closer to mpileup — multi-way pileup producing genotype likelihoods</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#Arguments-for-``bcftools-mpileup`-&lt;https://samtools.github.io/bcftools/bcftools.html#mpileup&gt;`__"><cite>Arguments for ``bcftools mpileup`</cite> &lt;https://samtools.github.io/bcftools/bcftools.html#mpileup&gt;`__</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#Manual:-How-to-call-the-variants?">Manual: How to call the variants?</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Some-further-references">Some further references</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#Call-SNP">Call SNP</a></li>
<li class="toctree-l2"><a class="reference internal" href="#View-VCF-files">View VCF files</a></li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Project: Calling SNPs in Yeast strain YJF153</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Call SNP using bcftools</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/34_call_snp.ipynb.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 5ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="Call-SNP-using-bcftools">
<h1>Call SNP using bcftools<a class="headerlink" href="#Call-SNP-using-bcftools" title="Permalink to this headline">¶</a></h1>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nb">source</span> config.sh
</pre></div>
</div>
</div>
<div class="section" id="bcftools-—-utilities-for-variant-calling-and-manipulating-VCFs-and-BCFs.">
<h2><a class="reference external" href="https://samtools.github.io/bcftools/bcftools.html">bcftools — utilities for variant calling and manipulating VCFs and BCFs.</a><a class="headerlink" href="#bcftools-—-utilities-for-variant-calling-and-manipulating-VCFs-and-BCFs." title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>annotate .. edit VCF files, add or remove annotations</p></li>
<li><p>call .. SNP/indel calling (former “view”)</p></li>
<li><p>cnv .. Copy Number Variation caller</p></li>
<li><p>concat .. concatenate VCF/BCF files from the same set of samples</p></li>
<li><p>consensus .. create consensus sequence by applying VCF variants</p></li>
<li><p>convert .. convert VCF/BCF to other formats and back</p></li>
<li><p>csq .. haplotype aware consequence caller</p></li>
<li><p>filter .. filter VCF/BCF files using fixed thresholds</p></li>
<li><p>gtcheck .. check sample concordance, detect sample swaps and contamination</p></li>
<li><p>index .. index VCF/BCF</p></li>
<li><p>isec .. intersections of VCF/BCF files</p></li>
<li><p>merge .. merge VCF/BCF files files from non-overlapping sample sets</p></li>
<li><p>mpileup .. multi-way pileup producing genotype likelihoods</p></li>
<li><p>norm .. normalize indels</p></li>
<li><p>plugin .. run user-defined plugin</p></li>
<li><p>polysomy .. detect contaminations and whole-chromosome aberrations</p></li>
<li><p>query .. transform VCF/BCF into user-defined formats</p></li>
<li><p>reheader .. modify VCF/BCF header, change sample names</p></li>
<li><p>roh .. identify runs of homo/auto-zygosity</p></li>
<li><p>sort .. sort VCF/BCF files</p></li>
<li><p>stats .. produce VCF/BCF stats (former vcfcheck)</p></li>
<li><p>view .. subset, filter and convert VCF and BCF files</p></li>
</ul>
<div class="section" id="bcftools-mpileup-is-used-when-calling-SNP">
<h3><strong>bcftools mpileup is used when calling SNP</strong><a class="headerlink" href="#bcftools-mpileup-is-used-when-calling-SNP" title="Permalink to this headline">¶</a></h3>
<p>When using <code class="docutils literal notranslate"><span class="pre">samtools</span> <span class="pre">mpileup</span></code>, the following error occurred: - <code class="docutils literal notranslate"><span class="pre">samtools</span> <span class="pre">mpileup</span> <span class="pre">-g</span> <span class="pre">-f</span> <span class="pre">*sorted.bam</span> <span class="pre">&gt;</span> <span class="pre">*.bcf</span></code></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>[warning] samtools mpileup option `g` is functional, but deprecated.
Please switch to using bcftools mpileup in future.
</pre></div>
</div>
<div class="line-block">
<div class="line">Therefore, instead of using samtools, here I use bcftools to call get <code class="docutils literal notranslate"><span class="pre">.bcf</span></code> file</div>
<div class="line">- <code class="docutils literal notranslate"><span class="pre">bcftools</span> <span class="pre">mpileup</span> <span class="pre">-f</span> <span class="pre">*sorted.bam</span> <span class="pre">&gt;</span> <span class="pre">*.bcf</span></code></div>
</div>
</div>
<div class="section" id="Look-closer-to-mpileup-—-multi-way-pileup-producing-genotype-likelihoods">
<h3><strong>Look closer to mpileup — multi-way pileup producing genotype likelihoods</strong><a class="headerlink" href="#Look-closer-to-mpileup-—-multi-way-pileup-producing-genotype-likelihoods" title="Permalink to this headline">¶</a></h3>
<p>Generate VCF or BCF containing genotype likelihoods for one or multiple alignment (BAM or CRAM) files. <strong>This is based on the original samtools mpileup command (with the -v or -g options)</strong> producing genotype likelihoods in VCF or BCF format, but not the textual pileup output. <strong>The mpileup command was transferred to bcftools in order to avoid errors resulting from use of incompatible versions of samtools and bcftools when using in the mpileup+bcftools call pipeline.</strong></p>
</div>
<div class="section" id="Arguments-for-``bcftools-mpileup`-&lt;https://samtools.github.io/bcftools/bcftools.html#mpileup&gt;`__">
<h3><cite>Arguments for ``bcftools mpileup`</cite> &lt;<a class="reference external" href="https://samtools.github.io/bcftools/bcftools.html#mpileup">https://samtools.github.io/bcftools/bcftools.html#mpileup</a>&gt;`__<a class="headerlink" href="#Arguments-for-``bcftools-mpileup`-<https://samtools.github.io/bcftools/bcftools.html#mpileup>`__" title="Permalink to this headline">¶</a></h3>
<p><em>Input options</em></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-f,</span> <span class="pre">--fasta-ref</span> <span class="pre">FILE</span></code></p>
<ul>
<li><p>The <strong>faidx-indexed reference</strong> file in the FASTA format.</p></li>
<li><p>The file can be optionally compressed by bgzip.</p></li>
<li><p><strong>Reference is required</strong> by default unless the –no-reference option is set [null]</p></li>
</ul>
</li>
</ul>
<p><em>Output options</em></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-O,</span> <span class="pre">--output-type</span> <span class="pre">b|u|z|v</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--threads</span> <span class="pre">INT</span></code></p></li>
</ul>
</div>
</div>
<div class="section" id="Manual:-How-to-call-the-variants?">
<h2><a class="reference external" href="http://samtools.github.io/bcftools/howtos/variant-calling.html">Manual: How to call the variants?</a><a class="headerlink" href="#Manual:-How-to-call-the-variants?" title="Permalink to this headline">¶</a></h2>
<p><code class="docutils literal notranslate"><span class="pre">bcftools</span> <span class="pre">mpileup</span> <span class="pre">-f</span> <span class="pre">reference.fa</span> <span class="pre">alignments.bam</span> <span class="pre">|</span> <span class="pre">bcftools</span> <span class="pre">call</span> <span class="pre">-mv</span> <span class="pre">-Ob</span> <span class="pre">-o</span> <span class="pre">calls.bcf</span></code> - The first <strong>mpileup</strong> part generates <strong>genotype likelihoods</strong> at each genomic position with coverage. - The second <strong>call</strong> part makes the actual calls. - The <code class="docutils literal notranslate"><span class="pre">-m</span></code> switch tells the program to use the default calling method, - the <code class="docutils literal notranslate"><span class="pre">-v</span></code> option asks to output only variant sites, - finally the <code class="docutils literal notranslate"><span class="pre">-O</span></code> option selects the output format. - In this example we chosen binary compressed BCF, which is the optimal
starting format for further processing, such as filtering.</p>
<p><strong>Fine tuning</strong></p>
<p>However, there are a few more options one should know about.</p>
<p><code class="docutils literal notranslate"><span class="pre">bcftools</span> <span class="pre">mpileup</span> <span class="pre">-Ou</span> <span class="pre">-f</span> <span class="pre">reference.fa</span> <span class="pre">alignments.bam</span> <span class="pre">|</span> <span class="pre">bcftools</span> <span class="pre">call</span> <span class="pre">-mv</span> <span class="pre">-Ob</span> <span class="pre">-o</span> <span class="pre">calls.bcf</span></code> - Do not waste computer’s time by making mpileup convert from the internal binary representation (BCF) to text (VCF), only to be immediately converted back to binary representation by call. Instead, use <code class="docutils literal notranslate"><span class="pre">-Ou</span></code> to work with uncompressed BCF output: - The calling can be made more sensitive or restrictive by using a different prior. Stricter calls are obtained by using smaller value, more benevolent calls
are obtained by using bigger value. The default is <code class="docutils literal notranslate"><span class="pre">bcftools</span> <span class="pre">call</span> <span class="pre">-P</span> <span class="pre">1.1e-3</span></code></p>
<hr class="docutils" />
<div class="section" id="Some-further-references">
<h3>Some further references<a class="headerlink" href="#Some-further-references" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>SNP tutorial: <a class="reference external" href="https://angus.readthedocs.io/en/2013/snp_tutorial.html">https://angus.readthedocs.io/en/2013/snp_tutorial.html</a></p></li>
<li><p>Bcftools manual: <a class="reference external" href="https://samtools.github.io/bcftools/">https://samtools.github.io/bcftools/</a></p></li>
<li><p>Question: samtools mpileup to bcftools mpileup: <a class="reference external" href="https://www.biostars.org/p/351305/">https://www.biostars.org/p/351305/</a></p></li>
<li><p>Manual</p>
<ul>
<li><p>older version:https://github.com/samtools/bcftools/wiki/HOWTOs</p></li>
<li><p>newer version: <a class="reference external" href="http://samtools.github.io/bcftools/howtos/index.html">http://samtools.github.io/bcftools/howtos/index.html</a></p></li>
</ul>
</li>
<li><p>Discussion in Github <a class="reference external" href="https://github.com/samtools/bcftools/issues/958">https://github.com/samtools/bcftools/issues/958</a></p></li>
</ul>
</div>
</div>
<hr class="docutils" />
<div class="section" id="Call-SNP">
<h2>Call SNP<a class="headerlink" href="#Call-SNP" title="Permalink to this headline">¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nv">DIR</span><span class="o">=(</span> <span class="s2">&quot;</span><span class="nv">$ALIGN_RAW</span><span class="s2">&quot;</span>  <span class="s2">&quot;</span><span class="nv">$ALIGN_TRIM</span><span class="s2">&quot;</span> <span class="s2">&quot;</span><span class="nv">$ALIGN_FILTER</span><span class="s2">&quot;</span> <span class="o">)</span>
<span class="nv">OUT</span><span class="o">=(</span> <span class="s2">&quot;</span><span class="nv">$OUT_RAW</span><span class="s2">&quot;</span>    <span class="s2">&quot;</span><span class="nv">$OUT_TRIM</span><span class="s2">&quot;</span>   <span class="s2">&quot;</span><span class="nv">$OUT_FILTER</span><span class="s2">&quot;</span> <span class="o">)</span>
<span class="nv">FIG</span><span class="o">=(</span> <span class="s2">&quot;</span><span class="nv">$FIG_RAW</span><span class="s2">&quot;</span>    <span class="s2">&quot;</span><span class="nv">$FIG_TRIM</span><span class="s2">&quot;</span>   <span class="s2">&quot;</span><span class="nv">$FIG_FILTER</span><span class="s2">&quot;</span> <span class="o">)</span>
<span class="nv">REFFA</span><span class="o">=</span><span class="nv">$GENOME</span>/<span class="nv">$FA</span>

<span class="k">for</span> <span class="o">((</span><span class="nv">i</span><span class="o">=</span><span class="m">0</span><span class="p">;</span> i&lt;<span class="si">${#</span><span class="nv">OUT</span><span class="p">[@]</span><span class="si">}</span><span class="p">;</span> ++i<span class="o">))</span><span class="p">;</span> <span class="k">do</span>
    <span class="nb">echo</span> <span class="s2">&quot;================================================&quot;</span>
    ls <span class="s2">&quot;</span><span class="si">${</span><span class="nv">DIR</span><span class="p">[i]</span><span class="si">}</span><span class="s2">&quot;</span>
    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">OUT</span><span class="p">[i]</span><span class="si">}</span><span class="s2">&quot;</span>
    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">FIG</span><span class="p">[i]</span><span class="si">}</span><span class="s2">&quot;</span>
    <span class="nb">echo</span> <span class="s2">&quot;++++++++++++++++++++++++&quot;</span>
<span class="k">done</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
================================================
aln_pe.bam  aln_pe_sorted.bam  calls_filter.stats
aln_pe.sam  calls.bcf          calls_filter.vcf
/home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_raw
/home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_raw/plot_stat
++++++++++++++++++++++++
================================================
aln_pe.bam  aln_pe.sam  aln_pe_sorted.bam
/home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_trim
/home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_trim/plot_stat
++++++++++++++++++++++++
================================================
aln_pe.bam  aln_pe.sam  aln_pe_sorted.bam
/home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_filter
/home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_filter/plot_stat
++++++++++++++++++++++++
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>bcftools mpileup --threads <span class="m">8</span> -Ou -f <span class="nv">$GENOME</span>/<span class="nv">$FA</span> <span class="nv">$ALIGN_RAW</span>/aln_pe_sorted.bam <span class="p">|</span> bcftools call -mv -Ob -o <span class="nv">$ALIGN_RAW</span>/calls.bcf
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid
[mpileup] 1 samples in 1 input files
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>bcftools view <span class="nv">$ALIGN_RAW</span>/calls.bcf <span class="p">|</span> bcftools filter -i<span class="s1">&#39;%QUAL&gt;20&#39;</span> &gt; <span class="nv">$ALIGN_RAW</span>/calls_filter.vcf
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>bcftools stats -F <span class="nv">$GENOME</span>/<span class="nv">$FA</span> -s - <span class="nv">$ALIGN_RAW</span>/calls_filter.vcf &gt; <span class="nv">$ALIGN_RAW</span>/calls_filter.stats
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>plot-vcfstats -p <span class="nv">$FIG_RAW</span> <span class="nv">$ALIGN_RAW</span>/calls_filter.stats
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Parsing bcftools stats output: /home/jovyan/work/Data/SRR4841864/align_raw/calls_filter.stats
Warning: Possible version mismatch, the definition line differs
        expected: # PSC [2]id   [3]sample       [4]nRefHom      [5]nNonRefHom   [6]nHets        [7]nTransitions [8]nTransversions       [9]nIndels      [10]average depth       [11]nSingletons
        found:    # PSC [2]id   [3]sample       [4]nRefHom      [5]nNonRefHom   [6]nHets        [7]nTransitions [8]nTransversions       [9]nIndels      [10]average depth       [11]nSingletons [12]nHapRef     [13]nHapAlt     [14]nMissing
Plotting graphs: python plot.py
Creating PDF: pdflatex summary.tex &gt;plot-vcfstats.log 2&gt;&amp;1
Finished: /home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_raw/plot_stat/summary.pdf
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span><span class="nv">DIR</span><span class="o">=(</span> <span class="s2">&quot;</span><span class="nv">$ALIGN_RAW</span><span class="s2">&quot;</span>  <span class="s2">&quot;</span><span class="nv">$ALIGN_TRIM</span><span class="s2">&quot;</span> <span class="s2">&quot;</span><span class="nv">$ALIGN_FILTER</span><span class="s2">&quot;</span> <span class="o">)</span>
<span class="nv">OUT</span><span class="o">=(</span> <span class="s2">&quot;</span><span class="nv">$OUT_RAW</span><span class="s2">&quot;</span>    <span class="s2">&quot;</span><span class="nv">$OUT_TRIM</span><span class="s2">&quot;</span>   <span class="s2">&quot;</span><span class="nv">$OUT_FILTER</span><span class="s2">&quot;</span> <span class="o">)</span>
<span class="nv">FIG</span><span class="o">=(</span> <span class="s2">&quot;</span><span class="nv">$FIG_RAW</span><span class="s2">&quot;</span>    <span class="s2">&quot;</span><span class="nv">$FIG_TRIM</span><span class="s2">&quot;</span>   <span class="s2">&quot;</span><span class="nv">$FIG_FILTER</span><span class="s2">&quot;</span> <span class="o">)</span>
<span class="nv">REFFA</span><span class="o">=</span><span class="nv">$GENOME</span>/<span class="nv">$FA</span>

<span class="k">for</span> <span class="o">((</span><span class="nv">i</span><span class="o">=</span><span class="m">0</span><span class="p">;</span> i&lt;<span class="si">${#</span><span class="nv">OUT</span><span class="p">[@]</span><span class="si">}</span><span class="p">;</span> ++i<span class="o">))</span><span class="p">;</span> <span class="k">do</span>
    <span class="nb">echo</span> <span class="s2">&quot;================================================&quot;</span>
    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">DIR</span><span class="p">[i]</span><span class="si">}</span><span class="s2">&quot;</span>
    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">OUT</span><span class="p">[i]</span><span class="si">}</span><span class="s2">&quot;</span>
    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">FIG</span><span class="p">[i]</span><span class="si">}</span><span class="s2">&quot;</span>
    <span class="nb">echo</span> <span class="s2">&quot;++++++++++++++++++++++++&quot;</span>

    <span class="c1">### Call SNP</span>
    bcftools mpileup --threads <span class="m">8</span> -Ou -f <span class="nv">$GENOME</span>/<span class="nv">$FA</span> <span class="si">${</span><span class="nv">DIR</span><span class="p">[i]</span><span class="si">}</span>/aln_pe_sorted.bam <span class="p">|</span> bcftools call -mv -Ob -o <span class="si">${</span><span class="nv">OUT</span><span class="p">[i]</span><span class="si">}</span>/calls.bcf

    <span class="c1">### Filter the variants</span>
    bcftools view <span class="si">${</span><span class="nv">OUT</span><span class="p">[i]</span><span class="si">}</span>/calls.bcf <span class="p">|</span> bcftools filter -i<span class="s1">&#39;%QUAL&gt;20&#39;</span> &gt; <span class="si">${</span><span class="nv">OUT</span><span class="p">[i]</span><span class="si">}</span>/calls_filter.vcf

    <span class="c1">### calculate stats of variants</span>
    bcftools stats -F <span class="nv">$GENOME</span>/<span class="nv">$FA</span> -s - <span class="si">${</span><span class="nv">OUT</span><span class="p">[i]</span><span class="si">}</span>/calls_filter.vcf &gt; <span class="si">${</span><span class="nv">OUT</span><span class="p">[i]</span><span class="si">}</span>/calls_filter.stats

    <span class="c1">### generate reports/plots of variants</span>
    plot-vcfstats -p <span class="si">${</span><span class="nv">FIG</span><span class="p">[i]</span><span class="si">}</span> <span class="si">${</span><span class="nv">OUT</span><span class="p">[i]</span><span class="si">}</span>/calls_filter.stats
<span class="k">done</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
================================================
/home/jovyan/work/Data/SRR4841864/align_raw
/home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_raw
/home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_raw/plot_stat
++++++++++++++++++++++++
Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid
[mpileup] 1 samples in 1 input files
Parsing bcftools stats output: /home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_raw/calls_filter.stats
Warning: Possible version mismatch, the definition line differs
        expected: # PSC [2]id   [3]sample       [4]nRefHom      [5]nNonRefHom   [6]nHets        [7]nTransitions [8]nTransversions       [9]nIndels      [10]average depth       [11]nSingletons
        found:    # PSC [2]id   [3]sample       [4]nRefHom      [5]nNonRefHom   [6]nHets        [7]nTransitions [8]nTransversions       [9]nIndels      [10]average depth       [11]nSingletons [12]nHapRef     [13]nHapAlt     [14]nMissing
Plotting graphs: python plot.py
Creating PDF: pdflatex summary.tex &gt;plot-vcfstats.log 2&gt;&amp;1
Finished: /home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_raw/plot_stat/summary.pdf
================================================
/home/jovyan/work/Data/SRR4841864/align_trim
/home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_trim
/home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_trim/plot_stat
++++++++++++++++++++++++
Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid
[mpileup] 1 samples in 1 input files
Parsing bcftools stats output: /home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_trim/calls_filter.stats
Warning: Possible version mismatch, the definition line differs
        expected: # PSC [2]id   [3]sample       [4]nRefHom      [5]nNonRefHom   [6]nHets        [7]nTransitions [8]nTransversions       [9]nIndels      [10]average depth       [11]nSingletons
        found:    # PSC [2]id   [3]sample       [4]nRefHom      [5]nNonRefHom   [6]nHets        [7]nTransitions [8]nTransversions       [9]nIndels      [10]average depth       [11]nSingletons [12]nHapRef     [13]nHapAlt     [14]nMissing
Plotting graphs: python plot.py
Creating PDF: pdflatex summary.tex &gt;plot-vcfstats.log 2&gt;&amp;1
Finished: /home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_trim/plot_stat/summary.pdf
================================================
/home/jovyan/work/Data/SRR4841864/align_filter
/home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_filter
/home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_filter/plot_stat
++++++++++++++++++++++++
Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid
[mpileup] 1 samples in 1 input files
Parsing bcftools stats output: /home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_filter/calls_filter.stats
Warning: Possible version mismatch, the definition line differs
        expected: # PSC [2]id   [3]sample       [4]nRefHom      [5]nNonRefHom   [6]nHets        [7]nTransitions [8]nTransversions       [9]nIndels      [10]average depth       [11]nSingletons
        found:    # PSC [2]id   [3]sample       [4]nRefHom      [5]nNonRefHom   [6]nHets        [7]nTransitions [8]nTransversions       [9]nIndels      [10]average depth       [11]nSingletons [12]nHapRef     [13]nHapAlt     [14]nMissing
Plotting graphs: python plot.py
Creating PDF: pdflatex summary.tex &gt;plot-vcfstats.log 2&gt;&amp;1
Finished: /home/jovyan/work/GitRepo/Duke_CBB520_HW2/out_filter/plot_stat/summary.pdf
</pre></div></div>
</div>
</div>
<div class="section" id="View-VCF-files">
<h2>View VCF files<a class="headerlink" href="#View-VCF-files" title="Permalink to this headline">¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[23]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>cat <span class="nv">$OUT_FILTER</span>/calls_filter.vcf <span class="p">|</span> egrep -v <span class="s1">&#39;##&#39;</span> <span class="p">|</span> head -10
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  /home/jovyan/work/Data/SRR4841864/align_filter/aln_pe_sorted.bam
I       101     .       G       A       23.434  PASS    DP=6;VDB=0.00698754;SGB=-0.590765;MQ0F=0.166667;AC=2;AN=2;DP4=0,0,5,0;MQ=12     GT:PL   1/1:53,15,0
I       136     .       G       A       25.3429 PASS    DP=13;VDB=0.209095;SGB=-0.636426;RPB=0.442128;MQB=0.924449;BQB=0.442128;MQ0F=0.153846;ICB=1;HOB=0.5;AC=1;AN=2;DP4=4,0,7,0;MQ=17 GT:PL   0/1:58,0,46
I       155     .       T       A       40.4021 PASS    DP=14;VDB=0.24063;SGB=-0.590765;RPB=0.6821;MQB=0.584657;MQSB=1;BQB=0.896474;MQ0F=0.142857;ICB=1;HOB=0.5;AC=1;AN=2;DP4=6,1,5,0;MQ=20     GT:PL   0/1:73,0,73
I       166     .       C       T       32.3756 PASS    DP=15;VDB=0.138245;SGB=-0.556411;RPB=0.444858;MQB=0.140858;MQSB=0.996974;BQB=0.105399;MQ0F=0.133333;ICB=1;HOB=0.5;AC=1;AN=2;DP4=7,3,4,0;MQ=18   GT:PL   0/1:65,0,81
I       177     .       G       C       46.2562 PASS    DP=10;VDB=0.0643982;SGB=-0.590765;RPB=0.450401;MQB=0.900802;MQSB=0.833333;BQB=0.900802;MQ0F=0.1;ICB=1;HOB=0.5;AC=1;AN=2;DP4=1,2,5,0;MQ=26       GT:PL   0/1:79,0,61
I       184     .       C       A       23.2506 PASS    DP=10;VDB=0.199299;SGB=-0.511536;RPB=0.462292;MQB=0.924584;MQSB=0.809011;BQB=0.809011;MQ0F=0.1;ICB=1;HOB=0.5;AC=1;AN=2;DP4=5,1,1,2;MQ=24        GT:PL   0/1:56,0,76
I       188     .       C       T       24.243  PASS    DP=10;VDB=0.148533;SGB=-0.511536;RPB=0.577865;MQB=0.924584;MQSB=0.809011;BQB=0.924584;MQ0F=0.1;ICB=1;HOB=0.5;AC=1;AN=2;DP4=5,1,1,2;MQ=24        GT:PL   0/1:57,0,78
I       192     .       C       T       31.3244 PASS    DP=10;VDB=0.0483573;SGB=-0.556411;RPB=0.868321;MQB=0.62023;MQSB=0.833333;BQB=0.992367;MQ0F=0.1;ICB=1;HOB=0.5;AC=1;AN=2;DP4=4,0,2,2;MQ=26        GT:PL   0/1:64,0,76
I       286     .       A       T       56      PASS    DP=8;VDB=0.0485232;SGB=-0.616816;MQSB=1;MQ0F=0;AC=2;AN=2;DP4=0,0,5,1;MQ=18      GT:PL   1/1:86,18,0
</pre></div></div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="33_sort_index.html" class="btn btn-neutral float-left" title="Sort and Index the alignment results" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>